nca/NetworkConfig/NetworkConfigQueryRunner.py

```diff
@@ -54,7 +56,11 @@ def compute_final_results(self, output_format):
                                sort_keys=False)
         else:
             output = '\n'.join(self.query_iterations_output)
-        return self.numerical_result, output, self.num_not_executed
+        if ExplTracker().is_active() and expl_nodes and (output_format == 'txt' or output_format == 'txt_no_fw_rules'):
+            expl_out = ExplTracker().explain(expl_nodes.split(','))
+        else:
+            expl_out = ''
+        return self.numerical_result, output+expl_out, self.num_not_executed
```

Comment:
why do we need both active/inactive mode for `ExplTracker` and `expl_nodes` from output configuration to check before adding explanation in the output? cannot it just rely on the `expl_nodes` value?

Reply:
The condition `(output_format == 'txt' or output_format == 'txt_no_fw_rules')` is actually the supported output formats for adding output explanation. I would add this as a method of `ExplTracker`.

Reply:
Can the call to compute and add the explanation output be done earlier? (connectivity query?) it is specific to the connectivity query, and the `compute_final_results` is a general function for all queries, so I find it confusing that it handles explanations added for a specific query usecase.

Reply:
> why do we need both active/inactive mode for `ExplTracker` and `expl_nodes` from output configuration to check before adding explanation in the output? cannot it just rely on the `expl_nodes` value?

This is for the cases when opt=false but --explain was given.



> The condition `(output_format == 'txt' or output_format == 'txt_no_fw_rules')` is actually the supported output formats for adding output explanation. I would add this as a method of `ExplTracker`.

Done. 



> Can the call to compute and add the explanation output be done earlier? (connectivity query?) it is specific to the connectivity query, and the `compute_final_results` is a general function for all queries, so I find it confusing that it handles explanations added for a specific query usecase.

Done.